### PR TITLE
Elixir factorial

### PR DESCRIPTION
**Pull request by cpackingham**
**Changes:**
Added factorial.exs

### CODE REVIEW

File: code/dynamic_programming/src/factorial/factorial.exs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+defmodule Factorial do
+  def factorial(0), do: 1
+  def factorial(1), do: 1
+  def factorial(num) do
+    num * factorial(num - 1)
+  end
+end
```
Comment (OvermindDL1) [inline code/dynamic_programming/src/factorial/factorial.exs]: There should be a newline at the end of the file as is traditional for Elixir.  Generally running the code through the built-in Elixir Formatter will format things in the 'proper' way.

**Comment by OvermindDL1** [comment]
@arnavb Good point, the last case should add a `when num > 1` to the gaurds.

File: code/dynamic_programming/src/factorial/factorial.exs (hunk 1 above)
Comment (cpackingham) [reply code/dynamic_programming/src/factorial/factorial.exs]: Elixir has a formatter?

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**